src/QtLocationPlugin/GenericMapProvider.cpp

```diff
@@ -8,6 +8,76 @@
  ****************************************************************************/
 #include "GenericMapProvider.h"
 
+static const QString JapanStdMapUrl = QStringLiteral("https://cyberjapandata.gsi.go.jp/xyz/std/%1/%2/%3.png");
+
+QString JapanStdMapProvider::_getURL(const int x, const int y, const int zoom, QNetworkAccessManager* networkManager) {
+    Q_UNUSED(networkManager)
+
+    int compulsionZoom = zoom;
+    if (compulsionZoom < 2) {
```

Comment:
The _getURL implementations should not filter the zoom level. If data is not available at that zoom level the requests should fail. The new master code in combination Qt Map implementation automatically handles missing tiles in way such that it will zoom the best tile it has to to higher zoom level for example.

I also believe that if you change the zoom level from what is requested then it will cache incorrect tile information in the database. You will end up with repeated tiles in the map.

Reply:
I would remove the ZOOM filter process.